skfda/exploratory/visualization/representation.py

```diff
@@ -333,143 +355,200 @@ def plot(
-                the samples with the same label are plotted in the same color.
-                If None, the default value, each sample is plotted in the color
-                assigned by matplotlib.pyplot.rcParams['axes.prop_cycle'].
-            group_colors: colors in which groups are represented,
-                there must be one for each group. If None, each
-                group is shown with distict colors in the "Greys" colormap.
-            group_names: name of each of the groups which appear
-                in a legend, there must be one for each one. Defaults to None
-                and the legend is not shown. Implies `legend=True`.
-            legend: if `True`, show a legend with the groups. If
-                `group_names` is passed, it will be used for finding the names
-                to display in the legend. Otherwise, the values passed to
-                `group` will be used.
-            kwargs: if dim_domain is 1, keyword arguments to be passed to
-                the matplotlib.pyplot.plot function; if dim_domain is 2,
-                keyword arguments to be passed to the
-                matplotlib.pyplot.plot_surface function.
-
-        Returns:
-            fig (figure object): figure object in which the graphs are plotted.
-
-        """
-        evaluated_points = None
+    ) -> None:
+        BasePlot.__init__(self)
+        self.fdata = fdata
+        self.grid_points = grid_points
 
+        self.evaluated_points = None
         if self.grid_points is None:
             # This can only be done for FDataGrid
             self.grid_points = self.fdata.grid_points
-            evaluated_points = self.fdata.data_matrix
+            self.evaluated_points = self.fdata.data_matrix
 
-        if evaluated_points is None:
-            evaluated_points = self.fdata(
+        if self.evaluated_points is None:
+            self.evaluated_points = self.fdata(
                 self.grid_points, grid=True,
             )
 
-        fig, axes = _get_figure_and_axes(chart, fig, ax)
-        fig, axes = _set_figure_layout_for_fdata(
-            self.fdata, fig, axes, n_rows, n_cols,
-        )
+        self.domain_range = domain_range
+        self.group = group
+        self.group_colors = group_colors
+        self.group_names = group_names
+        self.legend = legend
+
+        self._set_figure_and_axes(chart, fig, axes, n_rows, n_cols)
+
+    def plot(
+        self,
+        **kwargs: Any,
+    ) -> Figure:
+        """
+        Scatter FDataGrid object.
+
+        Args:
+        kwargs: if dim_domain is 1, keyword arguments to be passed to
+            the matplotlib.pyplot.plot function; if dim_domain is 2,
+            keyword arguments to be passed to the
+            matplotlib.pyplot.plot_surface function.
+        Returns:
+        fig: figure object in which the graphs are plotted.
+        """
+        self.artists = np.zeros(len(self.axes), dtype=np.ndarray)
```

Comment:
How bidimensional artist arrays work?

Reply:
I realised that as you can have more than one ax the we need a bidimensional array, it is updated in the multiple_disp pull request but basically we iterate the displays and afterwards the artists in it so that we can get all the artists instances